components/user_settings/sidebar/user_settings_sidebar.jsx

```diff
@@ -60,10 +60,10 @@ export default class UserSettingsSidebar extends React.Component {
     constructor(props) {
         super(props);
 
-        this.state = this.getStateFromStores();
+        this.state = this.getStateFromProps();
     }
 
-    getStateFromStores = () => {
+    getStateFromProps = () => {
```

Comment:
wouldn't be best to change this to be the new React lifecycle method `getDerivedStateFromProps`

Reply:
No, `getDerivedStateFromProps` runs before every single render and this should only update the state in certain cases.